artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3733,16 +3733,6 @@ private boolean deliver(final MessageReference ref) {
       }
    }
 
-   private static void accountForChangeInMemoryEstimate(final MessageReference ref, final int existingMemoryEstimate) {
-      final int delta = ref.getMessageMemoryEstimate() - existingMemoryEstimate;
-      if (delta > 0) {
```

Comment:
what if the delta was lower?

shouldn't you set messageMemoryEstimate, in case there are further changes on the message...

(just to keep it ready for future usages, say a transformation)

Reply:
it is quite specific to the case of decoding application properties which will always increase the usage.  it is currently the only operation that will cause the memoryEstimate to change.

Reply:
re worked fix to be less error prone via a direct reference to the owning page store in the message in stead of the message reference. https://github.com/apache/activemq-artemis/pull/3467